e2e/testcases/apiservice.bats

```diff
@@ -48,8 +63,11 @@ test_teardown() {
     acme/namespaces/resilient/namespace-resilient.yaml
   git::commit
 
+  debug::log "Adding APIService backend"
+  kubectl apply -f "${YAML_DIR}/namespace.yaml"
+  kubectl apply -f "${YAML_DIR}/rbac.yaml"
+  kubectl apply -f "${YAML_DIR}/namespace-custom-metrics.yaml"
```

Comment:
What is the intention of applying these resources manually? The purpose of this test case is to verify that config-sync mono repo mode can survive with a bad apiservice.

Reply:
The change in https://github.com/GoogleContainerTools/kpt-config-sync/pull/28 make errors like 503 blocking, where Config Sync will always retry syncing on bad APIs, thus previous test setup will timeout. Manually applying the deployment a while after APIService is deployed is to mock a temporary flaky API communication.

Reply:
Added comments to explain the manual application and cleanups during test teardown.